packages/valory/skills/decision_maker_abci/states/base.py

```diff
@@ -173,6 +172,29 @@ def claim(self) -> bool:
         """Get the claim."""
         return bool(self.db.get_strict("claim"))
 
+    @property
+    def mech_price(self) -> int:
+        """Get the mech's request price."""
+        return int(self.db.get_strict("mech_price"))
+
+    @property
+    def mech_requests(self) -> List[MechMetadata]:
+        """Get the mech requests."""
+        serialized = self.db.get("mech_requests", "[]")
+        if serialized is None:
+            serialized = "[]"
+        requests = json.loads(serialized)
+        return [MechMetadata(**metadata_item) for metadata_item in requests]
+
+    @property
+    def mech_responses(self) -> List[MechInteractionResponse]:
+        """Get the mech responses."""
+        serialized = self.db.get("mech_responses", "[]")
+        if serialized is None:
+            serialized = "[]"
+        responses = json.loads(serialized)
+        return [MechInteractionResponse(**response_item) for response_item in responses]
```

Comment:
Why redefined?

Reply:
Because these properties are set by the SynchronizedData of another ABCI app (mech_interact_abci), but they are read in the decision_maker_abci. If these are not redefined here, an AttributeError is launched because it does not find that attribute on the SynchronizedData. This approach has been mimicked from IEKit https://github.com/valory-xyz/IEKit/blob/main/packages/valory/skills/twitter_scoring_abci/rounds.py#L181-L194

Reply:
You can just inherit the synced data since there is already a dependency on the mech interact skill.

Reply:
Added a housekeeping issue to address it https://github.com/valory-xyz/trader/issues/243.